aten/src/ATen/mps/MPSDevice.mm

```diff
@@ -119,6 +119,11 @@ static inline MTLLanguageVersion getMetalLanguageVersion(const id<MTLDevice>& de
 
   static bool _macos_14_0_plus = [mpsCD instancesRespondToSelector:@selector(conjugateWithTensor:name:)] == YES;
 
+  // TODO: Change all version checks to use this API
```

Comment:
Have you checked that it actually works? Because NSProcessInfo will return version of the main app runtime, which targets MacOS 11 or something like that (and this is why I've used those horrible workarounds in the first place)

Reply:
That sounds concerning. I did do testing on this with a local MacOS15.0 and MacOS14.6 machines to verify that the API seems to be doing what asked during runtime and (after fiddling with the version numbers in the struct) is able to tell that one of my machines is below 15.1 and the other is below 14.7. 

But now that you mentioned let me also do a cross test of the API with compiling on one of the machines and checking the version on the other. I think we've previously been bitten by an API that on the surface looked like the correct way to verify the runtime os version but actually locked in the OS version during compile time leading to some interesting issues.

Reply:
I can verify that at least across these two OS major versions it seems to work as expected. Building the wheel and testing on a 14.6 machine I'm able to determine it is < 14.7. Copying over the same wheel to the 15.0 machine correctly determines it to be >= 14.7. So I'm feeling hopeful about the API being able to handle this so far.

After some snooping around the mentions of issues I found with the API seem to be related having an older SDK returning values that mismatch the current OS and around the MacOS11 time when the API was released there seems to have been some inconsistency how pre-11 and post-11 OS versions were handled by the NSProcessInfo.